### PR TITLE
Fix code quality issues

### PR DESCRIPTION
**Pull request by ravi-deepsource**
## Description

This pull request fixes some of the quality issues raised by [DeepSource](https://deepsource.io/gh/ravi-deepsource/chatwoot/) on my fork of this repository.

---
Fixes:
* Replaced some controllers to inherit from `ApplicationController` instead of `ActionController::Base`
* Updated conditional statement containing an assignment
* Simplified a conditional by replacing `unless bob.present?` with `if bob.blank?`
---

### Using DeepSource to continuously analyze your repository

1. Merge this PR. I have included a .deepsource.toml in this PR, which you can use to [configure](https://deepsource.io/docs/analyzer/ruby.html) your analysis settings.
2. Install DeepSource on your repository [here](https://deepsource.io/signup).
3. Activate analysis [here](https://deepsource.io/gh/chatwoot/chatwoot).

If you do not want to use DeepSource to continuously analyze this repo, I'll remove the `.deepsource.toml` from this PR and you can merge the rest of the fixes.

### CODE REVIEW

**Comment by azyzio** [comment]
I don't think those changes are good.

1. The `ApplicationCotnroller` is used for logged-in requests. `WidgetsController` is something completely different and should not inherit from `ApplicationController`
2. `marker = results.continuation_token.presence` is intended to be an assignment and not a comparison and the check is for existence

File: app/controllers/dashboard_controller.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-class DashboardController < ActionController::Base
+class DashboardController < ApplicationController
```
Comment (pranavrajs) [inline app/controllers/dashboard_controller.rb:1]: I guess this is intentional. The frontend is considered a separate app and it is not following the auth mechanism in application controller.

File: app/controllers/widget_tests_controller.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-class WidgetTestsController < ActionController::Base
+class WidgetTestsController < ApplicationController
```
Comment (pranavrajs) [inline app/controllers/widget_tests_controller.rb:1]: WidgetTests do not need to inherit ApplicationController for the reasons mentioned above

File: app/controllers/widgets_controller.rb [hunk 3]
```diff
@@ -1,4 +1,4 @@
-class WidgetsController < ActionController::Base
+class WidgetsController < ApplicationController
```
Comment (pranavrajs) [inline app/controllers/widgets_controller.rb:1]: Widget do not need to inherit ApplicationController for the reasons mentioned above

File: config/initializers/azure_storage_service_patch.rb [hunk 4]
```diff
@@ -73,7 +73,7 @@ def delete_prefixed(prefix)
             client.delete_blob(container, blob.name)
           end
 
-          break unless marker = results.continuation_token.presence
+          break unless marker == results.continuation_token.presence
```
Comment (pranavrajs) [inline config/initializers/azure_storage_service_patch.rb:76]: I'm not sure about this change @sojan-official please have a look.

**Comment by pranavrajs** [comment]
@ravi-deepsource I don't think the mentioned items are acutal issues. @sojan-official Please review

@azyzio I also feel changes are not good.

**Comment by nithindavid** [comment]
Hi @ravi-deepsource, Thanks for the efforts on this pr. But we don't feel like this is bringing any value to code, so we are closing this PR. 

About the deepsource integration, we will need to do look about it as we haven't used it before.

**Pull request closed without merging**